hw-model/src/lib.rs

```diff
@@ -476,77 +478,88 @@ pub trait HwModel {
     /// Create a model, and boot it to the point where CPU execution can
     /// occur. This includes programming the fuses, initializing the
     /// boot_fsm state machine, and (optionally) uploading firmware.
-    fn new(run_params: BootParams) -> Result<Self, Box<dyn Error>>
+    fn new(init_params: InitParams, boot_params: BootParams) -> Result<Self, Box<dyn Error>>
```

Comment:
Could the same be achieved without changing the function signature? That would remove the need to retouch all the tests.

Reply:
I definitely would have perferred to not change the API. The issue was the HW model combined the process of booting the HW model with the process of creating the model itself. This creates a verification coverage gap since you can't boot Caliptra multiple times without fully reinitializing the HW. This change separates those 2 processes to support that.

The alternative would be to leave init params as a member of boot params. That makes the API quite confusing since init is not part of boot anymore and those init parameters would then be ignored when calling boot().

Reply:
Thanks. that makes sense.